src/senaite/core/content/worksheet.py

```diff
@@ -0,0 +1,1556 @@
+                       "RejectAnalysis"),
+        multi_valued=False,
+        required=False,
+    )
+
+
+class IWorksheetSchema(model.Schema):
+    """Worksheet schema interface
+    """
+
+    title = schema.TextLine(
+        title=_(
+            u"title_worksheet_title",
+            default=u"Name"
+        ),
+        required=True,
+    )
+
+    description = schema.Text(
+        title=_(
+            u"title_worksheet_description",
+            default=u"Description"
+        ),
+        required=False,
+    )
+
+    worksheet_template = UIDReferenceField(
+        title=_(
+            u"title_worksheet_worksheettemplate",
+            default=u"Worksheet Template"
+        ),
+        allowed_types=("WorksheetTemplate",),
+        multi_valued=False,
+        required=False,
+    )
+
+    layout_view = DataGridField(
+        title=_(
+            u"title_worksheet_layout",
+            default=u"Layout"
+        ),
+        value_type=DataGridRow(schema=IWorksheetLayoutSchema),
+        default=[],
+        required=False,
+    )
+
+    # all layout info lives in Layout; Analyses is used for back references.
+    analyses = UIDReferenceField(
+        title=_(
+            u"title_worksheet_analyses",
+            default=u"Analyses"
+        ),
+        allowed_types=("Analysis", "DuplicateAnalysis", "ReferenceAnalysis",
+                       "RejectAnalysis"),
+        relationship="WorksheetAnalysis",
+        multi_valued=True,
+        required=True,
+    )
+
+    method = UIDReferenceField(
+        title=_(
+            u"title_worksheet_method",
+            default=u"Method"
+        ),
+        allowed_types=("Method",),
+        vocabulary="senaite.core.vocabularies.available_methods",
+        multi_valued=False,
+        required=False,
+    )
+
+    analyst = schema.TextLine(
+        title=_(
+            u"title_worksheet_analyst",
+            default=u"Analyst"
+        ),
+        required=False,
+        default=u""
+    )
+
+    # TODO Remove. Instruments must be assigned directly to each analysis.
+    instrument = UIDReferenceField(
+        title=_(
+            u"title_worksheet_method",
+            default=u"Method"
+        ),
+        allowed_types=("Instrument",),
+        vocabulary="senaite.core.vocabularies.available_methods",
+        multi_valued=False,
+        required=False,
+    )
+
+    directives.widget("remarks", RemarksWidget)
+    remarks = RemarksField(
+        title=_(
+            u"title_worksheet_remarks",
+            default=u"Remarks",
+        ),
+        required=False,
+    )
+
+    results_layout = schema.Choice(
+        title=_(
+            u"title_worksheet_results_layout",
+            default=u"Results Layout"
+        ),
+        vocabulary="senaite.core.vocabularies.worksheet_layouts",
+        required=False,
+        default=DEFAULT_WORKSHEET_LAYOUT,
+    )
+
+    replaced_by = UIDReferenceField(
```

Comment:
Remove this field. It is no longer necessary after the removal of the `reject` transition and related contents. See comment for `after_reject`